blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -248,11 +253,20 @@ private static Granularity granularityFromPointsLinear(double requestedDuration,
      *
      * @param requestedDuration (milliseconds)
      */
-    private static Granularity granularityFromPointsGeometric(double requestedDuration, int requestedPoints) {
+    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints) {
         double minimumPositivePointRatio = Double.MAX_VALUE;
         Granularity gran = null;
+        if (TTL_PROVIDER == null) {
+            TTL_PROVIDER = SafetyTtlProvider.getInstance();
+        }
 
         for (Granularity g : Granularity.granularities()) {
+            long ttl = getTTL(tenantid, g);
+
+            if (to < Calendar.getInstance().getTimeInMillis() - ttl) {
```

Comment:
The thing I'm concerned about is there may only be one or two points there and it will look wierd in the finder.
My preference is for the finder to look as good as possible even if it means we use less granular data.